gossipd/gossip.c

```diff
@@ -2375,6 +2463,11 @@ static struct io_plan *peer_disconnected(struct io_conn *conn,
 	status_trace("Forgetting remote peer %s",
 		     type_to_string(tmpctx, struct pubkey, &peer->id));
 
+	/* Disable any channels to and from this peer */
```

Comment:
And we should not do this if we this is caused by a reconnecting peer (below in this function).  Though I have a patch WIP which sets up a timer for every channel_update so we don't spam updates, which would also cover this.